doc/development_process/api_lifecycle.rst

```diff
@@ -0,0 +1,129 @@
+no longer optimal or supported by the underlying platforms.
+
+
+.. figure:: api_lifecycle.png
+    :align: center
+    :alt: API Life Cycle
+    :figclass: align-center
+
+    API Life Cycle
+
+
+Experimental
+*************
+
+Experimental APIs denote that a feature was introduced recently, and may change
+or be removed in future versions. Try it out and provide feedback
+to the community via the `Developer mailing list <https://lists.zephyrproject.org/g/devel>`_.
+
+Peripheral APIs (Hardware Related)
+==================================
+
+When introducing an API (header file with documentation) for a new peripheral
+or driver subsystem, review of the API is enforced and is driven by the API
+working group consisting of representatives from different vendors.
+
+The following requirements apply to new APIs:
+
+- Documentation of the API (Usage)
+  explaining its design and assumptions, how it is to be used, current
+  implementation limitations, and future potential, if appropriate.
+- The API introduction should be accompanied by at least one driver
+  implementing this API
+- At least one sample using the new API
+
+The API shall be promoted to ``unstable`` when it has at least two
+implementations on different hardware platforms.
+
+Hardware Agnostic APIs
+=======================
+
+For hardware agnostic APIs, multiple applications are required to promote an
+API from ``experimental`` to ``unstable``.
+
+
+Unstable
+********
+
+The API is in the process of settling, but has not yet had sufficient real-world
+testing to be considered stable. The API is considered generic in nature and can
+be used on diferrent hardware platforms.
+
+.. note::
+
+   Changes will not be announced.
+
+Stable
+*******
+
+The API has proven satisfactory, but cleanup in the underlying code may cause
+minor changes. Backwards-compatibility will be maintained if reasonable.
+
+An API can be declared ``stable`` after fullfilling the following requirements:
+
+- Test cases for the new API with 100% coverage
+- Complete documentation in code. All public interfaces shall be documentated
+  and available in online documentation.
+- The API has been in-use and was available in at least 2 development releases
+
+.. note::
+
+   Incompatible changes will be announced in the release notes.
+
+Deprecated
+***********
+
+.. note::
+
+   Unstable APIs can be removed without deprecation at any time.
+
+The following are the requirements for deprecating an existing API:
+
+- Deprecation Time (stable APIs): 2 Releases
+  The API needs to be marked as deprecated in at least two full releases
+  although in some cases, generally those out of our control, it might be
+  shorter than this. For example, if an API was first deprecated in 1.14,
+  it will be ready to be removed in 1.16 at the earliest.
+- What is required when deprecating:
+
+  - Mark as deprecated
+  - Document the deprecation
+  - All usage in Zephyr for deprecated API needs to be modified
```

Comment:
- Code using the deprecated API needs to be modified to remove usage of said API